pkg/reconciler/kubernetes/tektonresult/tektonresult.go

```diff
@@ -136,14 +146,149 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tr *v1alpha1.TektonResul
+			Get(ctx, existingInstallerSet, metav1.GetOptions{})
+		if err == nil {
+			tr.Status.MarkNotReady("Waiting for previous installer set to get deleted")
+			r.enqueueAfter(tr, 10*time.Second)
+			return nil
+		}
+		if !apierrors.IsNotFound(err) {
+			logger.Error("failed to get InstallerSet: %s", err)
+			return err
+		}
+		return nil
+
+	} else {
+		// If target namespace and version are not changed then check if spec
+		// of TektonResult is changed by checking hash stored as annotation on
+		// TektonInstallerSet with computing new hash of TektonResult Spec
+
+		// Hash of TektonResult Spec
+		expectedSpecHash, err := hash.Compute(tr.Spec)
+		if err != nil {
+			return err
+		}
+
+		// spec hash stored on installerSet
+		lastAppliedHash := installedTIS.GetAnnotations()[v1alpha1.LastAppliedHashKey]
+
+		if lastAppliedHash != expectedSpecHash {
+
+			if err := r.transform(ctx, &r.manifest, tr); err != nil {
+				logger.Error("manifest transformation failed:  ", err)
+				return err
+			}
+
+			// Update the spec hash
+			current := installedTIS.GetAnnotations()
+			current[v1alpha1.LastAppliedHashKey] = expectedSpecHash
+			installedTIS.SetAnnotations(current)
+
+			// Update the manifests
+			installedTIS.Spec.Manifests = r.manifest.Resources()
+
+			if _, err = r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+				Update(ctx, installedTIS, metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+
+			// after updating installer set enqueue after a duration
+			// to allow changes to get deployed
+			r.enqueueAfter(tr, 20*time.Second)
+			return nil
+		}
+	}
+
+	// Mark InstallerSetAvailable
+	tr.Status.MarkInstallerSetAvailable()
+
+	ready := installedTIS.Status.GetCondition(apis.ConditionReady)
+	if ready == nil {
+		tr.Status.MarkInstallerSetNotReady("Waiting for installation")
+		r.enqueueAfter(tr, 10*time.Second)
+		return nil
+	}
+
+	if ready.Status == corev1.ConditionUnknown {
+		tr.Status.MarkInstallerSetNotReady("Waiting for installation")
+		r.enqueueAfter(tr, 10*time.Second)
+		return nil
+	} else if ready.Status == corev1.ConditionFalse {
+		tr.Status.MarkInstallerSetNotReady(ready.Message)
+		r.enqueueAfter(tr, 10*time.Second)
+		return nil
+	}
+
+	// Mark InstallerSet Ready
+	tr.Status.MarkInstallerSetReady()
+
+	return nil
+}
+
+func (r *Reconciler) updateTektonResultsStatus(ctx context.Context, tr *v1alpha1.TektonResult, createdIs *v1alpha1.TektonInstallerSet) error {
+	// update the tr with TektonInstallerSet
+	tr.Status.SetTektonInstallerSet(createdIs.Name)
+
+	// Update the status with TektonInstallerSet so that any new thread
+	// reconciling with know that TektonInstallerSet is created otherwise
+	// there will be 2 instance created if we don't update status here
+	if _, err := r.operatorClientSet.OperatorV1alpha1().TektonResults().
+		UpdateStatus(ctx, tr, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+
+	return v1alpha1.RECONCILE_AGAIN_ERR
 }
```

Comment:
```suggestion
func (r *Reconciler) updateTektonResultsStatus(ctx context.Context, tr *v1alpha1.TektonResult, createdIs *v1alpha1.TektonInstallerSet) error {
	// update the tr with TektonInstallerSet
	tr.Status.SetTektonInstallerSet(createdIs.Name)

	return nil
```

this is enough as explicit status updates are an anti-pattern especially while using knative/pkg.
- the status update is handled in the generated code (in all knative/pkg based controllers). [example](https://github.com/tektoncd/operator/blob/a0ff713b69d769520e40f4629c41bcae63d85500/pkg/client/injection/reconciler/operator/v1alpha1/tektonpipeline/reconciler.go#L266)
- for more details refer to recording of last weeks operator wg call: [link](https://drive.google.com/file/d/1IhyctroGO6XKb8zdW2LET_UJqatUk7zJ/view?usp=sharing)

Reply:
the installerset's should be querried using labels anyway. so we are not dependent on this name in future reconcile loops.